[Identity] Prepare release for GA 4.0.0 (#27683)
Azure/azure-sdk-for-js: sdk/identity/identity/CHANGELOG.md

# Release History

> ## 4.0.0 (2023-11-07)

Reviewer: does Line 9 (Node 20 support) need to be revised at all?

Author: Node 20 support is there. But I think we on the sdk side have dropped the node 16 support. So i think this statement is still valid.

### Features Added

> - All the features shipped as part of 4.0.0-beta.1 will be GA with this version. The most important feature being the support of brokered authentication on Windows OS, such as WAM.

Author: ```suggestion
- All the features shipped as part of 4.0.0-beta.1 will be GA with this version. The most important features being the browser customization for success/ error messages and the support of brokered authentication on Windows OS, such as WAM.
```

# Release History

## 4.0.0 (2023-11-07)

### Features Added

> - All the features shipped as part of 4.0.0-beta.1 will be GA with this version. The most important features being the browser customization for success/ error messages and the support of brokered authentication on Windows OS, such as WAM.

Reviewer 2: ```suggestion
- All the features shipped as part of 4.0.0-beta.1 will be GA with this version. The most important features are the browser customization for success/error messages and the support of brokered authentication on Windows, such as WAM.
```
